src/components/modal/ConfirmModal/index.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { FC, useRef, useEffect } from 'react';
+
+import Overlay from '@/components/modal/Overlay';
+import {
+  modalWrapperStyle,
+  modalInnerStyle,
+  modalMessageStyle,
+  modalButtonWrapperStyle,
+  modalAcceptButtonStyle,
+  modalCancelButtonStyle,
+} from './style';
+
+type ConfirmModalProps = {
+  isOpen?: boolean;
+  message?: string;
+  cancelContent?: string;
+  acceptContent?: string;
+  cancelHandler?: () => void;
+  acceptHandler?: () => void;
+  closeHandler?: () => void;
+};
```

Comment:
atoms/confirmModalState.ts 의 ConfirmModalStateType과 중복인 요소가 있는 것 같습니다.
아래와 같이 바꿔보는 건 어떨까요?

```suggestion
type ConfirmModalProps = Partial<ConfirmModalStateType> &  {
  closeHandler?: () => void;
};
```

Reply:
맞습니당, 원래 `ConfirmModalStateType`을 export하고 있지 않아서 일일히 나열해주는 것을 선택했는 데 관련 hook 만들면서 export해주도록 수정해서... 해당 타입 사용해서 제안해주신대로 수정하도록 하겠습니다! 좋아요!!

Reply:
시도도 해보고 고민도 해봤는 데 기존의 방식이 더 낫다는 판단을 했습니다!

1. 제안해주신 대로 수정할 경우 `defaultProp "***" has no corresponding propTypes declaration`라는 오류가 발생합니다. => &로 합친 타입에 대해서 제대로 인식하지 못한 것 같습니다.
2. ConfirmModalStateType은 ConfirmModal 컴포넌트와는 **다른 context**를 가지고 있어서 의존성을 가지는 게 적절하지 않은 것 같습니다! ConfirmModal과 관련된 상태는 ConfirmModalRoot 컴포넌트에서 사용하고 있고 글로벌 상태를 변경함으로써 모달을 제어하기 위한 것입니다. 반면 현재 ConfirmModal 컴포넌트는 글로벌 상태와는 별개로 prop으로만 제어할 수 있습니다.

Reply:
@hseoy 
확인했습니다!! 가끔 코드 중복을 방지하고자 context를 소실시켜가며 무리하게 중복을 제거하는 경우가 있는데,
제 이번 리뷰가 그랬던 것 같아 죄송한 마음입니다 ㅠㅠ 좋습니다!!